### PR TITLE
Select - Hide placeholder

### PR DESCRIPTION
**Pull request by mayurikini**
### Summary
This code adds the ability for a consumer to hide `- Select -` placeholder.

### Additional Details
Added site examples and tests for following scenarios:
- when placeholder is hidden
- when placeholder is hidden and select is in invalid state (Here default will be blank until an option is selected)

This code is a solution for #1178 

Deployment link:
https://terra-core-deployed-pr-1229.herokuapp.com/static/#/site/select

### CODE REVIEW

File: packages/terra-form-select/src/Select.jsx [hunk 1]
```diff
@@ -300,14 +325,7 @@ class Select extends React.Component {
           boundingRef={boundingRef}
         >
           <SelectMenu>
-            <SelectOption
```
Comment (dylan-sftwr) [inline packages/terra-form-select/src/Select.jsx]: Can we remove the entire assignment on line 300 and simply do the evaluation as follows?
```jsx
<SelectMenu>
  { !isPlaceholderHidden &&
    <SelectOption
       value={defaultOptionValue}
       display={defaultOptionDisplay}
       isSelected={isDefaultSelected}
       onKeyDown={this.wrapOnKeyDown(defaultOptionValue)}
       onClick={this.wrapOnClick(defaultOptionValue)}
       className={cx('default-option')
    />
  }
  .
  .
  .
```

File: packages/terra-form-select/tests/nightwatch/select/BlankSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        isInvalid
+        isPlaceholderHidden
+        id="blankSelect"
+      >
+        <Select.Option value="puppies" display="Puppies" key="puppies" />
+        <Select.Option value="kittens" display="Kittens" key="kittens" />
+        <Select.Option value="snappers" display="Snappers" key="snappers" />
+        <Select.Option value="bumblers" display="Bumblers" disabled key="bumblers" />
+        <Select.Option value="joeys" display="Joeys" key="joeys" />
+        <Select.Option value="micros" display="Microprocessors" disabled key="miros" />
```
Comment (dylan-sftwr) [inline packages/terra-form-select/tests/nightwatch/select/BlankSelect.jsx]: typo `key="micros"`

File: packages/terra-form-select/tests/nightwatch/select/HiddenPlaceholderSelect.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      >
+        <Select.Option value="puppies" display="Puppies" key="puppies" />
+        <Select.Option value="kittens" display="Kittens" key="kittens" id="kittens" />
+        <Select.Option value="snappers" display="Snappers" key="snappers" />
+        <Select.Option value="bumblers" display="Bumblers" disabled key="bumblers" id="disabledOption" />
+        <Select.Option value="joeys" display="Joeys" key="joeys" />
+        <Select.Option value="micros" display="Microprocessors" disabled key="miros" />
+      </Select>
+    </form>
+    <br />
```
Comment (dylan-sftwr) [inline packages/terra-form-select/tests/nightwatch/select/HiddenPlaceholderSelect.jsx]: Same comments here regarding `miros` typo.

File: packages/terra-site/src/examples/form-select/BlankSelect.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            isInvalid
+            isPlaceholderHidden
+            onChange={this.handleChange}
+          >
+            <Select.Option value="puppies" display="Puppies" key="puppies" />
+            <Select.Option value="kittens" display="Kittens" key="kittens" />
+            <Select.Option value="snappers" display="Snappers" key="snappers" />
+            <Select.Option value="bumblers" display="Bumblers" disabled key="bumblers" />
+            <Select.Option value="joeys" display="Joeys" key="joeys" />
+            <Select.Option value="micros" display="Microprocessors" disabled key="miros" />
```
Comment (dylan-sftwr) [inline packages/terra-site/src/examples/form-select/BlankSelect.jsx]: typo `key="micros"`

File: packages/terra-site/src/examples/form-select/HiddenPlaceholderSelect.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            required
+            onChange={this.handleChange}
+            isPlaceholderHidden
+          >
+            <Select.Option value="puppies" display="Puppies" key="puppies" />
+            <Select.Option value="kittens" display="Kittens" key="kittens" />
+            <Select.Option value="snappers" display="Snappers" key="snappers" />
+            <Select.Option value="bumblers" display="Bumblers" disabled key="bumblers" />
+            <Select.Option value="joeys" display="Joeys" key="joeys" />
+            <Select.Option value="micros" display="Microprocessors" disabled key="miros" />
```
Comment (dylan-sftwr) [inline packages/terra-site/src/examples/form-select/HiddenPlaceholderSelect.jsx]: typo `key="micros"`

File: packages/terra-form-select/tests/nightwatch/select/SelectTests.jsx [hunk 6]
```diff
@@ -9,6 +9,8 @@ const SelectTests = () => (
       <li><Link to="/tests/select-tests/default">Select - Default</Link></li>
       <li><Link to="/tests/select-tests/disabled">Select - Disabled</Link></li>
       <li><Link to="/tests/select-tests/invalid">Select - Invalid</Link></li>
+      <li><Link to="/tests/select-tests/hidden-placeholder">Select - Hidden placeholder</Link></li>
```
Comment (dylan-sftwr) [inline packages/terra-form-select/tests/nightwatch/select/SelectTests.jsx]: nit: can you capitalize `placeholder`?

**Comment by neilpfeiffer** [comment]
Tested: PR #1229
Resolves: Issue #1178

\
**Reviewed:**
- enhancement matches approved adjustments to terra-form-select


\
\
Verified @neilpfeiffer, Ready for Release.

**Pass** |
-----------|
UX Review|

**Comment by mmalaker** [comment]
Functional verification completed. The form select examples display and function correctly with the option to hide or show the placeholder.

**Pull request merged**